### ISSUE TITLE
tests/test_app.py:66: AssertionError : assert 404 == 200 +  where 404 = <Response [404 NOT FOUND]>.status_code

### ISSUE THREAD

**Issue by giterinhub** [Azure-Samples/azure-search-openai-demo#1416]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [X] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
1. Create new Github Codespace of [latest version](https://github.com/Azure-Samples/azure-search-openai-demo/commit/d896376069d3350dc733638ae86e811beba53d9f)
2. Install the development dependencies:
```
python3 -m pip install -r requirements-dev.txt
```
3. Install the pre-commit hooks:
```
pre-commit install
```
4. Run the tests:
```
python3 -m pytest
```

### Any log messages given by the failure
```
_________________________________________________________________________________________________________________________________________ test_index[client0] __________________________________________________________________________________________________________________________________________

client = <quart.testing.client.QuartClient object at 0x7f835b037110>

    @pytest.mark.asyncio
    async def test_index(client):
        response = await client.get("/")
>       assert response.status_code == 200
E       assert 404 == 200
E        +  where 404 = <Response [404 NOT FOUND]>.status_code

tests/test_app.py:66: AssertionError
```

### Expected/desired behavior
>All tests should pass

### OS and Version?
> ```Linux codespaces-6836e2 6.2.0-1019-azure #19~22.04.1-Ubuntu SMP Wed Jan 10 22:57:03 UTC 2024 x86_64 GNU/Linux```

### azd version?
> azd version 1.7.0 (commit 49d6adc2efb178083f61822e6b4715258560803d)

### Versions
>N/A

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
Interesting, the tests all pass in the CI and I've been running the tests all day. I'll try the steps in Codespace to see if something's going awry there.

**Comment by pamelafox** [comment]
Ah, did you compile the Javascript before running the tests? The build step copies them into app/backend.

See what the workflow does here:
https://github.com/Azure-Samples/azure-search-openai-demo/blob/d896376069d3350dc733638ae86e811beba53d9f/.github/workflows/python-test.yaml#L44

We can add that to CONTRIBUTING.md, PRs welcome!

**Comment by giterinhub** [comment]
dear @pamelafox your feedback is spot-on, thanks a lot. I just created PR #1427 to update CONTRIBUTING.md. Could you give it a review? After compiling the JS, the tests pass, so my issue can be closed.
Enjoyed your talk at the Python Datascience π-Day! 😎

**Comment by giterinhub** [comment]
Resolved through PR #1425 - much appreciated!
